include/Gaffer/Set.h

```diff
@@ -90,6 +103,44 @@ class GAFFER_API Set : public IECore::RunTimeTyped, public boost::signals::track
 
 IE_CORE_DECLAREPTR( Set );
 
+template<typename ContainerType, typename ValueType>
+class SetIterator : public boost::iterator_facade<SetIterator<ContainerType, ValueType>, ValueType, boost::forward_traversal_tag>
```

Comment:
Nice! This is a great addition. Can we go further and make it a random access iterator rather than just forward_traversal? Or at least make it bidirectional?

There are also a few little things to fix :

- Indentation is off everywhere, again.
- Naming convention dictates Iterator and ConstIterator rather than iterator and const_iterator. I don't think this should affect the use with range-for should it? That just cares about begin/end?
- `#include "boost/...` not `#include <boost/`.

Reply:
> Can we go further and make it a random access iterator rather than just forward_traversal? Or at least make it bidirectional?

Do you want me to split it off into its own PR if we do that?

> I don't think this should affect the use with range-for should it? That just cares about begin/end?

Yeah, range-for only cares about begin/end as far as I'm aware. I'll rename them.

> Indentation is off everywhere, again.

Sorry, I've tweaked my editor settings now to hopefully finally take care of this.

One more thing: I thought it's not very elegant to templatize the iterator on two parameters just so that the constness of `Member` and `Set` match. I couldn't quite figure out how to infer one from the other, though. Maybe doing it this way is alright.